interview-questions.md

```diff
@@ -14,44 +14,67 @@ Researched answer:
+
+Github is a version control source system that works as remote repository for collborators. It allows for developers to work on a project. We can build up different parts of a project at the same time using duplicate branches. We can then merge changes to the main branch, while also being able to revert those changes.
+
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: Logical/comparison/equality 
+Comparison
+>
+<
+<=
+>=
+
+Logical AND && 
+Logical OR ||
+Logical Not !
+
+Equality
+=== Strict equality
+== Loose equality
 
 Researched answer:
 
 3. What is an index? What is the difference between index and value?
 
-Your answer:
+Your answer: In an Array each value has a location called the index. The value is the data located at the index inside of an Array. 
 
 Researched answer:
+Indexes are numbers that give us a location or value that is being stored at that location. They are are zero-indexed meaning they begin counting from 0 and not 1. We can use this to access information in a string or array. In order to access them we must use the bracket notation []. You can input the value which is the data located at the index or the index (0,1,2, etc.) which is the location.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Repeating a certain criteria until it is met.
 
 Researched answer:
+In Javascript Loops an iteration is used in statements where a process is repeated a number of times until a condition is met.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
 Your answer:
+There is a navigator and driver. The driver will be the one typing the code and will be told what to do by the navigator. The navigator is responsible for making sure the instructions are being said to the driver. They navigator will essentially build up the code but isn't the one to type it.
 
-Researched answer:
-
+Researched answer: 
+In pair programming there is a driver and navigator. A driver and and navigator work together to build code. The driver will listen to instructions give by the navigator and produce the code. In order for the two to be efficient they must communicate effectively. This can mean building a plan together and making sure there is a understanding of what is to be done. Working with another person creates efficiency with the knowledge of both people. It also helps with keeping focus on the task as both people should be engaged. But a driver should not be the one to code on their own with no guidance. There should be respect between the two of proper workload balance. If there is an issue of not knowing what to for the navigator, they will have a driver to properly work through it with.
 ## Looking Ahead: Terms for Next Week
 
```

Comment:
Having some context when hearing the terms during lectures makes learning to code a bit easier.